sketch_oct9a.ino

```diff
@@ -119,160 +121,152 @@ void updateSDCardLEDs() {
 }
 
 void setup() {
-  Serial.begin(9600);
+  Serial1.begin(9600);
 
+#if CAPSULE == 2
   // Pin A6: analog input from VOC sensor
   pinMode(A6, PinMode::INPUT);
+#endif
   // Pin D5: SD LEDs
   pinMode(5, PinMode::OUTPUT);
   // Pin D6: GPS LEDs
   pinMode(6, PinMode::OUTPUT);
   // Pin D7: sensor LEDs
   pinMode(7, PinMode::OUTPUT);
 
-#ifdef _SERCOM_CLASS_
   pinPeripheral(0, PIO_SERCOM);
   pinPeripheral(1, PIO_SERCOM);
   pinPeripheral(8, PIO_SERCOM);
   pinPeripheral(9, PIO_SERCOM);
+  pinPeripheral(A1, PIO_SERCOM);
+  pinPeripheral(A2, PIO_SERCOM);
   pinPeripheral(A3, PIO_SERCOM);
   pinPeripheral(A4, PIO_SERCOM);
-#endif
 
   // Turn on the error LEDs until initialization finishes
   digitalWrite(5, LOW);
   digitalWrite(6, LOW);
   digitalWrite(7, LOW);
 
-  while (!Serial) { /* wait for serial port to connect */ }
-
   // Initialize sensors
   alt.initialize();
   imu.initialize();
 #if CAPSULE == 2
   hum.initialize();
 #endif
   updateSensorLEDs();
-}
 
-void printAcceleration(const IMU::vector3& accel) {
-  Serial.print('(');
-  Serial.print(accel.x);
-  Serial.print(", ");
-  Serial.print(accel.y);
-  Serial.print(", ");
-  Serial.print(accel.z);
-  Serial.print(") - magnitude ");
-
-  float magnitude = IMU::getMagnitude(accel);
-  Serial.print(magnitude);
+  card.initialize();
+  updateSDCardLEDs();
+
+  gps.initialize();
+  updateGPSLEDs();
+
+  Scheduler.startLoop(gps_loop);
+  Scheduler.startLoop(radio_loop);
 }
 
-void printGPSData(const GPS::Coordinates& loc) {
-  if (loc.latitude < 0) {
-    Serial.print(-loc.latitude);
-    Serial.print("S ");
-  } else {
-    Serial.print(loc.latitude);
-    Serial.print("N ");
+void loop() {
+#if CAPSULE == 2
+  last_voc = analogRead(6);
+
+  if (hum.getStatus() != HumiditySensor::ACTIVE) {
+    hum.initialize();
+    updateSensorLEDs();
   }
-  if (loc.longitude < 0) {
-    Serial.print(-loc.longitude);
-    Serial.print('W');
-  } else {
-    Serial.print(loc.longitude);
-    Serial.print('E');
+
+  float humid, temp;
+  if (hum.getValues(&humid, &temp)) {
```

Comment:
Why not immediately use 'last_humid' and 'last_temp'? Similar question for all instances like these.

Reply:
I'll double-check the implementation, but I believe that if `getValues` returns false, the values that you passed pointers to have undefined values rather than being left as-is. If this turns out to be wrong I'll change it as per your suggestion, and if not I'll make it more clear in comments.